pkg/controller/disaggregated_cluster_controller.go

```diff
@@ -202,52 +202,52 @@ func (dc *DisaggregatedClusterReconciler) Reconcile(ctx context.Context, req rec
 	hv := hash.HashObject(ddc.Spec)
 
 	var res ctrl.Result
-	////TODO: deprecated.
-	//cmnn := types.NamespacedName{Namespace: ddc.Namespace, Name: ddc.Spec.InstanceConfigMap}
-	//ddcnn := types.NamespacedName{Namespace: ddc.Namespace, Name: ddc.Name}
-	//cmnnStr := cmnn.String()
-	//ddcnnStr := ddcnn.String()
-	//if _, ok := dc.wcms[cmnnStr]; !ok {
-	//	dc.wcms[cmnnStr] = ddcnnStr
-	//}
-
-	//sync resource.
-	//recall all errors
 	var msg string
 	reconRes, reconErr := dc.reconcileSub(ctx, &ddc)
 	if reconErr != nil {
 		msg = msg + reconErr.Error()
+	}
+	if !reconRes.IsZero() {
 		res = reconRes
 	}
 
 	// clear unused resources.
 	clearRes, clearErr := dc.clearUnusedResources(ctx, &ddc)
 	if clearErr != nil {
 		msg = msg + reconErr.Error()
+	}
+
+	if !clearRes.IsZero() {
 		res = clearRes
 	}
 
 	//display new status.
 	disRes, disErr := func() (ctrl.Result, error) {
 		//reorganize status.
-		if res, err = dc.reorganizeStatus(&ddc); err != nil {
-			return res, err
+		var stsRes ctrl.Result
+		var stsErr error
+		if stsRes, stsErr = dc.reorganizeStatus(&ddc); stsErr != nil {
+			return stsRes, stsErr
 		}
 
 		//update cr or status
-		if res, err = dc.updateObjectORStatus(ctx, &ddc, hv); err != nil {
-			return res, err
+		if stsRes, stsErr = dc.updateObjectORStatus(ctx, &ddc, hv); stsErr != nil {
+			return stsRes, stsErr
 		}
 
-		return ctrl.Result{}, nil
+		return stsRes, stsErr
 	}()
-
 	if disErr != nil {
 		msg = msg + disErr.Error()
+	}
+	if !disRes.IsZero() {
```

Comment:
This use mode is not right, please learn the difference between struct call and pointer call.